pkg/resources/helpers.go

```diff
@@ -130,6 +130,35 @@ func GetPropertyAsPointer[T any](d *schema.ResourceData, property string) *T {
 	return &typedValue
 }
 
+func GetFirstNestedObjectByKey[T any](d *schema.ResourceData, propertyKey string, nestedValueKey string) *T {
```

Comment:
Yeah, I was thinking about returning an error and I guess I left it as is. I'll change it so it will return an error as a second return value with meaningful errors.